### PR TITLE
Bug 1841119: apply extendedArguments directly to kcm in bootstrap

### PR DESCRIPTION
**Pull request by soltysh**
Follow-up to #471 which missed the bootstrap phase.

/assign @tnozicka

### CODE REVIEW

**Comment by soltysh** [comment]
Unrelated failures
/retest

**Review by tnozicka — COMMENTED** [review]
lgtm, a unit for `kube-controller-manager-pod.yaml` rendering would be nice

File: pkg/cmd/render/render_test.go [hunk 1]
```diff
@@ -194,10 +240,60 @@ func TestRenderCommand(t *testing.T) {
...
+				if err != nil {
+					t.Errorf("error reading file %s: %v", p, err)
+					continue
+				}
+				dataJSON, err := yaml.YAMLToJSON(data)
+				if err != nil {
+					t.Errorf("error converting file %s: %v", p, err)
+					continue
+				}
+				configObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, dataJSON)
```
Comment (tnozicka) [inline pkg/cmd/render/render_test.go:254]: `scheme.Codecs.UniversalDeserializer()`?

Comment (tnozicka) [reply pkg/cmd/render/render_test.go:254]: the yaml dance feels weird for deserialization but I am not sure if there is a generic one for unstructured

**Review by tnozicka — COMMENTED** [review]
/lgtm

**Pull request merged**